File: tests/astro/python/test_julian_date.py
#!/usr/bin/env python3

import astropy.time
import datetime

year   = 2000
month  = 1
day    = 1
hour   = 0
minute = 0
second = 0
dt = datetime.datetime(year, month ,day, hour, minute, second)
time = astropy.time.Time(dt)
print(time.jd);

year   = 2024
month  = 4
day    = 24
hour   = 17
minute = 15
second = 30
dt = datetime.datetime(year, month ,day, hour, minute, second)
time = astropy.time.Time(dt)
print(time.jd);

File: tests/astro/TestJulianDate.cpp
#include <gtest/gtest.h>

#include <mcutils/astro/JulianDate.h>

class TestJulianDate : public ::testing::Test
{
protected:
    TestJulianDate() {}
    virtual ~TestJulianDate() {}
    void SetUp() override {}
    void TearDown() override {}
};

TEST_F(TestJulianDate, CanCalculateJulianDay)
{
    // expected values calculated with astropy
    // tests/astro/python/test_julian_date.py

    mc::DateTime dt;
    mc::JulianDate jd;

    dt.year   = 2000;
    dt.month  = 1;
    dt.day    = 1;
    dt.hour   = 0;
    dt.minute = 0;
    dt.second = 0;
    dt.msec   = 0;
    jd.SetFromGregorianDate(dt);
    EXPECT_DOUBLE_EQ(jd.jd(), 2451544.5);

    dt.year   = 2024;
    dt.month  = 4;
    dt.day    = 24;
    dt.hour   = 17;
    dt.minute = 15;
    dt.second = 30;
    dt.msec   = 0;
    jd.SetFromGregorianDate(dt);
    EXPECT_DOUBLE_EQ(jd.jd(), 2460425.219097222);
}
